gcloud-java-dns/src/test/java/com/google/cloud/dns/it/ITDnsTest.java

```diff
@@ -967,25 +967,920 @@ public void testListDnsRecords() {
+          Dns.RecordSetListOption.fields(Dns.RecordSetField.NAME));
+      DnsBatchResult<Page<RecordSet>> recordsResult = batch.listRecordSets(zone.name(),
+          Dns.RecordSetListOption.fields(Dns.RecordSetField.DNS_RECORDS));
+      DnsBatchResult<Page<RecordSet>> typeResult = batch.listRecordSets(zone.name(),
+          Dns.RecordSetListOption.fields(Dns.RecordSetField.TYPE),
+          Dns.RecordSetListOption.pageSize(1)); // also paging
+      DnsBatchResult<Page<RecordSet>> pageSizeResult = batch.listRecordSets(zone.name(),
+          Dns.RecordSetListOption.fields(Dns.RecordSetField.TYPE),
+          Dns.RecordSetListOption.pageSize(1));
+      batch.submit();
+      Iterator<RecordSet> recordSetIterator = ttlResult.get().iterateAll();
+      int counter = 0;
+      while (recordSetIterator.hasNext()) {
+        RecordSet recordSet = recordSetIterator.next();
+        assertEquals(recordSets.get(counter).ttl(), recordSet.ttl());
+        assertEquals(recordSets.get(counter).name(), recordSet.name());
+        assertEquals(recordSets.get(counter).type(), recordSet.type());
+        assertTrue(recordSet.records().isEmpty());
+        counter++;
+      }
+      assertEquals(2, counter);
+      recordSetIterator = nameResult.get().iterateAll();
+      counter = 0;
+      while (recordSetIterator.hasNext()) {
+        RecordSet recordSet = recordSetIterator.next();
+        assertEquals(recordSets.get(counter).name(), recordSet.name());
+        assertEquals(recordSets.get(counter).type(), recordSet.type());
+        assertTrue(recordSet.records().isEmpty());
+        assertNull(recordSet.ttl());
+        counter++;
+      }
+      assertEquals(2, counter);
+      recordSetIterator = recordsResult.get().iterateAll();
+      counter = 0;
+      while (recordSetIterator.hasNext()) {
+        RecordSet recordSet = recordSetIterator.next();
+        assertEquals(recordSets.get(counter).records(), recordSet.records());
+        assertEquals(recordSets.get(counter).name(), recordSet.name());
+        assertEquals(recordSets.get(counter).type(), recordSet.type());
+        assertNull(recordSet.ttl());
+        counter++;
+      }
+      assertEquals(2, counter);
+      recordSetIterator = typeResult.get().iterateAll(); // also test paging
+      counter = 0;
+      while (recordSetIterator.hasNext()) {
+        RecordSet recordSet = recordSetIterator.next();
+        assertEquals(recordSets.get(counter).type(), recordSet.type());
+        assertEquals(recordSets.get(counter).name(), recordSet.name());
+        assertTrue(recordSet.records().isEmpty());
+        assertNull(recordSet.ttl());
+        counter++;
+      }
+      assertEquals(2, counter);
+      // test page size
+      Page<RecordSet> recordSetPage = pageSizeResult.get();
+      assertEquals(1, ImmutableList.copyOf(recordSetPage.values().iterator()).size());
+      // test name filter
+      ChangeRequest change = DNS.applyChangeRequest(ZONE1.name(), CHANGE_ADD_ZONE1);
```

Comment:
The point is that the default records, which have the same name, are not present when the filter is used.